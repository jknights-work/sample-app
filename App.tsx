/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 * @flow
 */

import React, { Component } from 'react';
import { View } from 'react-native';

/* Common */
import { Props } from '@common/component/common.props';
import { AppStore } from './modules/common/event/store/app.store'; 

/* Redux */
import { Provider } from 'react-redux';

/** Components */
import MainComponent from './src/main/main.component';
import SlidesComponent from './src/main/components/intro/slides.component';
import SettingsComponent from './src/main/components/settings/settings.component';
import HeaderComponent from './src/main/components/header/header.component';

/** Register Services */

/** Navigation */
import { Router, Stack, Scene, Drawer } from 'react-native-router-flux';
import HeaderMenuComponent from './src/main/components/header/header.menu.component';

import { Icon } from 'native-base';



export default class App extends Component<Props> {

  componentDidMount () {}

  noBackButton () {
   // return (  )
  }

  render() {
    return (
      <Provider store={AppStore}>
      
          <Router>
              <Stack key="root" >
              
              
              <Drawer
                    drawerIcon={<Icon name="menu"/>}
                    drawer
                    hideNavBar
                    key="drawer"
                    drawerWidth={300}
                    contentComponent={HeaderMenuComponent}  
              >
                    <Scene key="home" component={MainComponent} title="WUWIN" panHandlers={null} intial renderBackButton={()=> <View/>} />
                    <Scene key="splash" component={SlidesComponent} />
                    <Scene key="settings" component={SettingsComponent} navBar={HeaderComponent} title="Settings" swipeEnabled renderBackButton={() => <Icon name="back"/>} />
                    <Scene key="side" component={MainComponent} title="Menu" />
                </Drawer>
             </Stack>
          </Router>
      </Provider>
    );
  }
}

