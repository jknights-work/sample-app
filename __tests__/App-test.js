// __tests__/App-test.js
import React from 'react';
import App from '../app';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
