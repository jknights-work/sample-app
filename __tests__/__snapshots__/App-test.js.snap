// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Change init
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      false
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Location : 
      false
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Get Location
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Current Location : 
      
    </Text>
  </View>
</View>
`;
